crates/isograph_compiler/src/write_artifacts.rs

```diff
@@ -59,6 +63,49 @@ pub(crate) fn write_artifacts_to_disk(
                 message: e.to_string(),
             })?;
     }
+
+    if let Some(pd_config) = config.options.persisted_documents.as_ref() {
```

Comment:
Also just commenting. If the user changes the config, we might end up in a situation where the old persisted json was written at `/1.json`, and now is written at `/2.json`, and both end up created (i.e. #1 is not deleted). That's fine!

We should eventually keep track of all the generated artifacts that we know about from the previous build and diff that with the new set of generated artifacts. So this problem will solve itself.